External/Plugins/ProjectManager/Actions/FileActions.cs

```diff
@@ -446,33 +446,31 @@ public bool Rename(string oldPath, string newName)
                 if (isDirectory)
                 {
                     // this is required for renaming directories, don't ask me why
-                    string oldPathFixed = (oldPath.EndsWith('\\')) ? oldPath : oldPath + "\\";
-                    string newPathFixed = (newPath.EndsWith('\\')) ? newPath : newPath + "\\";
-                    if (oldPathFixed.Equals(newPathFixed, StringComparison.OrdinalIgnoreCase))
+                    if (oldPath.EndsWith('\\')) oldPath = oldPath.Substring(0, oldPath.Length - 1);
+                    if (newPath.EndsWith('\\')) newPath = newPath.Substring(0, newPath.Length - 1);
+                    if (oldPath.Equals(newPath, StringComparison.OrdinalIgnoreCase))
                     {
                         // name casing changed
-                        string tmpPath = newPathFixed.Substring(0, newPathFixed.Length - 1) + "$renaming$\\";
-                        Directory.Move(oldPathFixed, tmpPath);
-                        oldPathFixed = tmpPath;
+                        string tmpPath = newPath + "$renaming$";
+                        Directory.Move(oldPath, tmpPath);
+                        Directory.Move(tmpPath, newPath);
                     }
-                    if (FileHelper.ConfirmOverwrite(newPath))
+                    else if (FileHelper.ConfirmOverwrite(newPath))
                     {
-                        FileHelper.ForceMoveDirectory(oldPathFixed, newPathFixed);
+                        FileHelper.ForceMoveDirectory(oldPath, newPath);
                     }
                     else return false;
                 }
                 else
                 {
-                    string oldName = Path.GetFileName(oldPath);
-                    if (oldName.Equals(newName, StringComparison.OrdinalIgnoreCase))
+                    if (oldPath.Equals(newPath, StringComparison.OrdinalIgnoreCase))
                     {
                         // name casing changed
                         string tmpPath = newPath + "$renaming$";
                         File.Move(oldPath, tmpPath);
-                        oldPath = tmpPath;
+                        File.Move(tmpPath, newPath);
```

Comment:
But if that's the case, the `else` block is also not proper, as `FileHelper.ForceMove()` just does `File.Move()`.
Aren't `CodeRefactor.Commands.Rename` and `RenameFile` called in the cases where documents are renamed? I think this bit of code is like generic part, like for miscellaneous files.
Sadly I don't have a computer (gg rip) to run through method references so I can't confirm now.